### ISSUE TITLE
More usable html reports : logging the requests and responses in the report and naming entries

### ISSUE THREAD

**Issue by christopheblin** [Orange-OpenSource/hurl#1884]
### Problem to solve

When you execute multiple tests, you can use --error-format long to have the response of the failed assertion

However, it is often useful to have the full context of the failure, mostly the body of the request

For ex, if you use a captured variable of a previous request as an input to the failing request, seeing only something like "expected status 200 but was 400" and the body with (hopefully) an error can be insuficient

So you have to find the logs of the test run in order to find the failing request

And sometimes, in order to understand why the payload was like it was, you also need to look at previous calls

### Proposal

If the command contains --vey-verbose, then the html report should contain the request AND the response of all entries (or another option "--report-format long")

With this, a better UI would be needed (collapsed sub sections for certificate, request headers, request body, response header, response body for example)

As an additional hint, it would also be great to be able to name the entries

For ex, 

```
GET https://acmecorp.net/info as Retrieve info

HTTP 200
```

would display "Retrieve info" isntead of "Entry 1"

That would allow to have a far more usable reports compared to

![image](https://github.com/Orange-OpenSource/hurl/assets/11044877/58d0ccef-5fc1-4fb9-ba11-cedccb9b414f)

### Drawabacks

The size of the report is going to be far more important which could be a problem in some circumstances (very long script with lots of HTTP requests)

**Comment by jcamiel** [comment]
Hi @christopheblin reviewing this issue, I made the link with:

- Issue [include response bodies in the json output](https://github.com/Orange-OpenSource/hurl/issues/1907)
- Discussion [Can the response body be included when using the --json flag?](https://github.com/Orange-OpenSource/hurl/discussions/1969)

These issue/discussion were about including response bodies when using `--json`. It makes sense when using `--json` and I think it also make sense in the HTML report (even if there are no errors). My only fear is the size of the report and the "performance" hit of dumping all responses to disk. 

For `--json` we're going to include it by default. Just need to decide if we dump response (and display it) in th HTML report by default or behind an option.

**Comment by jcamiel** [comment]
Hi @christopheblin regarding the size of the HTML report, which can be an issue, we proposed to intoduce a new option `--max-body` which will be the maximum number of bytes for a request / response body to be included in the HTML report. We could also reuse this option for `--json`:

```
--max-body <COUNT>
          Maximum number of bytes for exporting a body, -1 for unlimited size [default: 256]
```

If you want the bodies to be always saved:

```shell 
$ hurl --max-body -1 --report-html /tmp/report.html *.hurl 
```

If you want the bodies to be never saved:

```shell 
$ hurl --max-body 0 --report-html /tmp/report.html *.hurl 
```

Or we a fixed limitation:

```shell 
$ hurl --max-body 128000 --report-html /tmp/report.html *.hurl 
```

**Comment by simonzkl** [comment]
Very interested in this as well. It would really help with debugging failing tests without having to pollute output with `--very-verbose` or trying to piece together requests manually with JSON reports. The `--max-body` approach to limit bloat seems like a fine way to do it.

**Comment by simonzkl** [comment]
I hope you don't mind a bit of a meta-bug report, but I wanted to sum up what where I think Hurl is currently lacking in terms of test debuggability. It's unnecessarily difficult to piece together what went wrong in a Hurl test. There is not one reporting mode that ticks all the check boxes:
- HTML reports don't contain actual requests and response bodies (#1884)
- JSON reports contain responses, but not requests. They're split up into files and named by request uuid, which makes them difficult to stitch together without scripting. But even if you could, you'd be missing request bodies (#1907)
- The `--very-verbose` flag is the only mode that has all the request/response information, but it pollutes the main stdout/stderr output so it's not a great mode to use in CI by default. This is what I usually use to debug test failures. Not because it's great, but because there's no alternative.

**Comment by jcamiel** [comment]
Hi @simonzkl thanks for your comment, we've not forgot it, it's "simply" always a matter of priority!
Also note that, instead of `--very-verbose`, you can use `--error-format long` to only log the failings requests

**Comment by simonzkl** [comment]
Yep, aware of this flag. But sometimes you also need the context from previous requests. Thanks!
